lib/active_merchant/billing/gateways/fortis.rb

```diff
@@ -0,0 +1,330 @@
+module ActiveMerchant # :nodoc:
+  module Billing # :nodoc:
+    class FortisGateway < Gateway
+      self.test_url = 'https://api.sandbox.fortis.tech/v1'
+      self.live_url = 'https://api.fortis.tech/v1'
+
+      self.supported_countries = %w{US CA}
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover jbc unionpay]
+      self.money_format = :cents
+      self.homepage_url = 'https://fortispay.com'
+      self.display_name = 'Fortis'
+
+      STATUS_MAPPING = {
+        101 => 'Sale cc Approved',
+        102 => 'Sale cc AuthOnly',
+        111 => 'Refund cc Refunded',
+        121 => 'Credit/Debit/Refund cc AvsOnly',
+        131 => 'Credit/Debit/Refund ach Pending Origination',
+        132 => 'Credit/Debit/Refund ach Originating',
+        133 => 'Credit/Debit/Refund ach Originated',
+        134 => 'Credit/Debit/Refund ach Settled',
+        191 => 'Settled (deprecated - batches are now settled on the /v2/transactionbatches endpoint)',
+        201 => 'All cc/ach Voided',
+        301 => 'All cc/ach Declined',
+        331 => 'Credit/Debit/Refund ach Charged Back'
+      }
+
+      REASON_MAPPING = {
```

Comment:
was the mapping for status and reason needed due the Fortis response does not return a readable message, or was it for another specific  reason ❓

Reply:
Most of the times Fortis returns an status_mapping and a reason_mapping but not the actual message